packages/features/bookings/lib/handleNewBooking/ensureAvailableUsers.ts

```diff
@@ -121,6 +122,81 @@ const _ensureAvailableUsers = async (
     },
   });
 
+  if (eventType.restrictionScheduleId) {
+    try {
```

Comment:
Is it possible to move try catch into a new function:
```
if (eventType.restrictionScheduleId) {
    await checkRestrictionSchedule(eventType.restrictionScheduleId);
}
```

Also, I don't understand - the catch block just logs the error even if we throw `throw new Error(ErrorCode.NoAvailableUsersFound);` - shouldn't we return early in this case? Why log the caught error and proceed with code execution?

Reply:
Essentially we are doing these checks but if code fails we just proceed with execution?